src/utils/clm_time_manager.F90

```diff
@@ -1759,6 +1760,44 @@ end function is_perpetual
 
   !=========================================================================================
 
+  logical function is_doy_in_interval(start, end, doy_in)
+
+    ! Return true if day of year is in the provided interval.
+    ! Does not treat leap years differently from normal years.
+    ! Arguments
+    integer, intent(in) :: start ! start of interval (day of year)
+    integer, intent(in) :: end ! end of interval (day of year)
+    integer, optional, intent(in) :: doy_in ! day of year to query
```

Comment:
- [ ] Do you anticipate using the 'optional' aspect here? If just "maybe", I'd suggest making this required because having it optional makes it possible to accidentally misuse the function (forgetting to specify `doy_in`, but it compiles and runs... but does the wrong thing). If yes, then please add some documentation up here in the argument list saying what happens if doy_in isn't provided. But actually, maybe a less error-prone way to handle this would be: make doy_in required, then introduce a new function, `is_today_in_interval(start, end)` which simply calls `is_doy_in_interval(start, end, get_prev_calday())`

Reply:
Good thinking! I added (7b57111) a new function, `is_today_in_doy_interval(start, end)` and am now requiring the third argument to `is_doy_in_interval()`.